packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -69,6 +80,22 @@ export class Cluster extends cdk.Construct implements ICluster {
     this.clusterName = cluster.clusterName;
   }
 
+  /**
+   * Add a CloudMap namespace for this cluster. Defaults to HTTP Namespace unless otherwise specified. FIXME: allow
+   * DNSNamespaces ?
+   */
+  public addNamespace(name: string): cloudmap.PrivateDnsNamespace {
+    const sdNamespace = new cloudmap.PrivateDnsNamespace(this, 'DefaultServiceDiscoveryNamespace', {
+      name,
```

Comment:
Is it possible to have this autogenerated?

Does it make sense to default it to some particular value that we can derive from another value?

Does it make sense to add a boolean to the constructor props to call this automatically?

Reply:
Have name generated? No, that is usually the one thing customers would want to specify -- it's the domain name, so like `foo.com`

not sure what you mean by the boolean in the constructor props?

Reply:
I was thinking:

```ts
class ClusterProps {
   // ...
   /**
    * Domain name for the private DNS namespace to create
    *
    * @default No namespace is created by default
    */
   privateDnsNamespaceDomainName?: string
}
```

Which would imply a call to `addNamespace`.

But looking at this, I do have a couple more questions:

* Would you ever have more than one namespace? (Right now you can't call this more than once)
* Would it always be a private namespace? (Right now doesn't look like you could do a public or HTTP namespace even if you wanted to)
* Is `cluster.addNamespace()` descriptive enough? (Maybe `cluster.addCloudMapNamespace()?`).